sdl_android/src/main/java/com/smartdevicelink/api/audio/AudioStreamManager.java

```diff
@@ -0,0 +1,271 @@
+ * to the connected device. Audio files can be pushed to the manager in order to
+ * play them on the connected device. The manager uses the Android built-in MediaCodec.
+ */
+@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
+public class AudioStreamManager extends BaseSubManager {
+    private static final String TAG = AudioStreamManager.class.getSimpleName();
+
+    private IAudioStreamListener sdlAudioStream;
+    private int sdlSampleRate;
+    private @SampleType int sdlSampleType;
+    private final Queue<BaseAudioDecoder> queue;
+
+    private StreamingStateMachine streamingStateMachine;
+
+    // INTERNAL INTERFACE
+
+    private final ISdlServiceListener serviceListener = new ISdlServiceListener() {
+        @Override
+        public void onServiceStarted(SdlSession session, SessionType type, boolean isEncrypted) {
+            if (SessionType.PCM.equals(type)) {
+                sdlAudioStream = session.startAudioStream();
+                streamingStateMachine.transitionToState(StreamingStateMachine.STARTED);
+            }
+        }
+
+        @Override
+        public void onServiceEnded(SdlSession session, SessionType type) {
+            if (SessionType.PCM.equals(type)) {
+                if (internalInterface != null) {
+                    session.stopAudioStream();
+                    sdlAudioStream = null;
+                    internalInterface.removeServiceListener(SessionType.PCM, this);
+                    streamingStateMachine.transitionToState(StreamingStateMachine.NONE);
+                }
+            }
+        }
+
+        @Override
+        public void onServiceError(SdlSession session, SessionType type, String reason) {
+            Log.e(TAG, "OnServiceError: " + reason);
+        }
+    };
+
+    /**
+     * Creates a new object of AudioStreamManager
+     * @param internalInterface The internal interface to the connected device.
+     */
+    public AudioStreamManager(@NonNull ISdl internalInterface) {
+        super(internalInterface);
+        this.queue = new LinkedList<>();
+
+        streamingStateMachine = new StreamingStateMachine();
+        transitionToState(BaseSubManager.READY);
+    }
+
+    @Override
+    public void dispose() {
+        stopAudioStream();
+
+        streamingStateMachine.transitionToState(StreamingStateMachine.NONE);
+    }
+
+    /**
+     * Starts the audio service and audio stream to the connected device.
+     * The method is non-blocking.
+     * @param encrypted Specify whether or not the audio stream should be encrypted.
+     */
+    public void startAudioStream(boolean encrypted) {
+        // audio stream cannot be started without a connected internal interface
+        if (internalInterface == null || !internalInterface.isConnected()) {
+            return;
+        }
+
+        // streaming state must be NONE (starting the service is ready. starting stream is started)
+        if (streamingStateMachine.getState() != StreamingStateMachine.NONE) {

```

Comment:
I believe this logic is already built into the streamingStateMachine. It will only change state on a valid transition (NONE->READY or NONE->ERROR) so this check probably isn't necessary